source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -154,80 +145,38 @@ namespace aspect
-      return transformed;
-    }
-
     template <>
     void
     EllipsoidalChunk<3>::create_coarse_mesh(parallel::distributed::Triangulation<3> &coarse_grid) const
     {
       const int dim = 3;
-      //std::cout << "semi_major_axis_a: " <<  manifold.semi_major_axis_a << ":" << manifold.get_semi_major_axis_a() << std::endl;
-      const Point<3> corner_points[2] = { Point<3>(corners_rectangle[0][0]*numbers::PI/180,
-                                                   corners_rectangle[0][1]*numbers::PI/180,
-                                                   -bottom_depth),
-                                          Point<3>(corners_rectangle[2][0]*numbers::PI/180,
-                                                   corners_rectangle[2][1]*numbers::PI/180,
-                                                   0)
-                                        };
-      std::vector<unsigned int> subdivisions(3);
-      subdivisions[0] = EW_subdiv; // divide east-west
-      subdivisions[1] = NS_subdiv; // divide north-south length
-      subdivisions[2] = depth_subdiv; // divide depth
-      GridGenerator::subdivided_hyper_rectangle (coarse_grid, subdivisions,
-                                                 corner_points[0], corner_points[1],
-                                                 true);
-      if (para_to_rect_angle != 0)
-        {
-          // transform rectangle back to parallelogram
-          GridTools::transform (std_cxx11::bind(&EllipsoidalChunkGeometry::rotate_rectangle_to_parallelogram,
-                                                std_cxx11::cref(manifold),
-                                                std_cxx11::_1),
-                                coarse_grid);
-        }
-      if (rot_para_to_para_angle != 0)
-        {
-          // transform parallelogram back to rotated parallelogram
-          GridTools::transform (std_cxx11::bind(&EllipsoidalChunkGeometry::rotate_parallelogram,
-                                                std_cxx11::cref(manifold),
-                                                std_cxx11::_1),
-                                coarse_grid);
-        }
-      //transform to the ellipsoid surface
+
+      /**
+       * Generate parallelepiped grid with one point (point 0) at (0,0,0) and the
+       * other corners (respectivly corner 1,2 and 4) placed relative to that point.
+       */
+      const Point<3> corner_points[dim] = {Point<dim>((corners[1][0]-corners[0][0])*numbers::PI/180,
+                                                      (corners[1][1]-corners[0][1])*numbers::PI/180,
+                                                      0),
+                                           Point<dim>((corners[3][0]-corners[0][0])*numbers::PI/180,
+                                                      (corners[3][1]-corners[0][1])*numbers::PI/180,

```

Comment:
should this be corners[3] - corners[2]? Or does it not matter, because corners[0] and corners[2] are on the same latitude/longitude?